File: 169. Majority Element/Java/Main.java
class Solution {

    public int majorityElement(int[] nums) {
        int count = 0;
        int candidate = nums[0];

        for (int item : nums) {
            if (count == 0) {
                candidate = item;
            }
            count += (item == candidate) ? 1 : -1;
        }

        return candidate;
    }
}

public class Main {
    public static void main(String[] args) {
        Solution sol = new Solution();
        int[] arr1 = { 3, 2, 3 };
        System.out.print("[3,2,3] = ");
        System.out.println(sol.majorityElement(arr1)); // Output: 3

        int[] arr2 = { 2, 2, 1, 1, 1, 2, 2 };
        System.out.print("[2,2,1,1,1,2,2] = ");
        System.out.println(sol.majorityElement(arr2)); // Output: 2
    }
}

File: 169. Majority Element/JS/Main.js
var majorityElement = function(nums) {
    let count = 0;
    let candidate=null;
    for(item of nums){
        if(count === 0){
            candidate = item;
        }
        count+= item === candidate?1:-1;
        console.log("Item", item);
    }
    return candidate;
};

console.log("[3,2,3] = ", majorityElement([3,2,3]));

console.log("[2,2,1,1,1,2,2] = ", majorityElement([2,2,1,1,1,2,2]));